docs/V1/TokenizationAndCompilation.md

```diff
@@ -0,0 +1,96 @@
+# Tokenization and Compilation
+This document summarizes the tokenization and compilation process that is at the core of
+Buphalo's algorithm to convert a template into a new file.
+
+## Process Overview
+For each `actor` in a `fablet`, Buphalo performs the following:
+1. Runs Annotation Processors to replace Annotations
+   - Annotation Processors may replace their annotation with text that includes:
+      - Text that will get replaced by a token according to the tokenization rules
+      - A token that will get replaced during compilation
+      - Text that won't be modified by tokenization/compilation
+2. Replaces pre-defined strings with Tokens (Tokenization)
+3. Replaces each Token with a new string based on the file (Compilation)
+
+## Annotation Processors
+See [Annotation Processors][APs].
+
+## Example Values
+Examples derived based on the single actor in the following file when the
+`Neighborhoods_Buphalo_V1_TargetApplication_BuilderInterface__NamespacePrefix` env variable is set to
+`Neighborhoods\BuphaloFitness\Demo3`.
+
+```yml
+# src/DOR/V1/Car.buphalo.v1.fabrication.yml
+<PrimaryActorName>/Builder/Factory.php
+  template: PrimaryActorName/Builder/SpecialFactory.php
+```
+
+## Tokenization
+Note that in the Regex/String column, {} represents a dynamic string:
+* {Template File Name}:
+  Represents a call to `$this->getActorTemplate()->getFabricationFileActor()->getTemplateFileName()`
+  e.g. `SpecialFactory` 
+* {Actor Short Pascal Case Name}:
+  Represents a call to `$this->getActor()->getShortPascalCaseName()`
+  e.g. `Factory`
+
+|Type|Regex/String|Replacement|
+|---|---|---|
+|Regex|`namespace(\s+)Neighborhoods\\BuphaloTemplateTree\\PrimaryActorName`|`namespace **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`namespace(\s+)Neighborhoods\\BuphaloTemplateTree`|`namespace **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**`|
+|Regex|`use(\s+)Neighborhoods\\BuphaloTemplateTree\\PrimaryActorName`|`use **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`use(\s+)Neighborhoods\\BuphaloTemplateTree/`|`use **NAMESPACE_PREFIX_TOKEN**`|
```

Comment:
is the trailing slash here intentional?

Reply:
ah looks like you fixed it in [a subsequent PR](https://github.com/neighborhoods/Buphalo/pull/68/files#diff-39f8fcffc170dcbb86a02d267df321c4701f8de8733317ab9cff1b9c7858c063L43)